src/java.base/share/classes/java/io/Reader.java

```diff
@@ -189,14 +189,17 @@ protected Reader(Object lock) {
      * Attempts to read characters into the specified character buffer.
      * The buffer is used as a repository of characters as-is: the only
      * changes made are the results of a put operation. No flipping or
-     * rewinding of the buffer is performed.
+     * rewinding of the buffer is performed.  If the specified character
+     * buffer is full, then no characters will be read and zero will be
```

Comment:
Instead of "is full" here, it would be more consistent to state "has no remaining elements".

Reply:
For a buffer that is being written into, it sounds a bit odd to say "no remaining elements"; that phrasing is ok for reading/getting bytes but doesn't sound like there's free space to put anything.

Would "no remaining free elements" or "no remaining available elements" be better?

Reply:
Or "no remaining space" / "no space remaining"?

Reply:
I have a slight preference for "no space remaining"; remaining can still be a linkplain to the method if that helps it be specific.

Reply:
I concur. Thanks.

Reply:
Addressed in 055107bddcc38a331ee93e178402473ab9f91740.